src/client.ts

```diff
@@ -797,19 +797,27 @@ export class SpvWalletClient {
     const privateKeys: bsv.PrivateKey[] = []
     draftTransaction.configuration.inputs?.forEach((input, index) => {
       if (input.destination) {
-        const chainKey = xPriv.deriveChild(input.destination.chain)
-        const numKey = chainKey.deriveChild(input.destination.num)
-        privateKeys.push(numKey.privateKey)
-
-        // small sanity check for the inputs
-        if (
-          input.transaction_id != txDraft.inputs[index].prevTxId.toString('hex') ||
-          input.output_index != txDraft.inputs[index].outputIndex
-        ) {
-          const Err = new Error('input tx ids do not match in draft and transaction hex')
-          logger.error(Err)
-          throw Err
+        const dst = input.destination
+        // derive private key (m/chain/num)
+        let privKey =xPriv.deriveChild(dst.chain)
+                          .deriveChild(dst.num)
+
+        if (dst.paymail_external_derivation_num){
```

Comment:
Because the `paymail_external_derivation_num` has type `number | undefined`, your `if` condition would not work as intended for `0` value. So, I guess it's better to check `!= null` which checks both `undefined` and `null`.
```suggestion
        if (dst.paymail_external_derivation_num != null){
```